internal/oauthtoken/file.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 EngFlow Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauthtoken
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/oauth2"
+)
+
+// FileStore persists tokens in JSON files in a directory, one file per
+// token. The tokens are stored unecrypted. This implementation may be used
+// when Keyring cannot be used, for example, in a CI environment when there is
+// no encrypted keyring.
+type FileStore struct {
+	dir string
+}
+
+var _ LoadStorer = (*FileStore)(nil)
+
+func NewFileTokenStore(dir string) (*FileStore, error) {
+	if err := os.MkdirAll(dir, 0700); err != nil {
```

Comment:
nit: Looks like the way this is used, we expect it to only ever create the final leaf directory - but in the weird case that it doesn't, it'd be unfortunate to create a directory chain with `0700`, no? Maybe doing `os.Mkdir` and ignoring `fs.ErrExist` would more pedantically reflect our expectations

...is it also worthwhile to ensure an existing directory carries the correct permission bits? Looks like `MkdirAll` won't error or change permissions if the dirs already exist (if someone can pre-create them as `0777` before running `engflow_auth`, this will gladly write tokens to a globally-readable/editable dir!)

Reply:
Good point. I think we do have to create the parent directories, but they should not be `0700`. I've changed that to `0777`, with `0700` only for the leaf directory.

About checking permissions after the directory is created: permission bits are a Unix-only concept and aren't supported on all file systems, so that check would be unreliable. Windows has a read-only bit, but most permissions are controlled through ACLs, which I don't want to attempt to mess with there.